### DISCUSSION TITLE
Support "passive" mode

### DISCUSSION THREAD

**Discussion by gitkumi** [TanStack/query#5156] [Ideas] [answered]
From the docs:

```
Out of the box, TanStack Query is configured with aggressive but sane defaults. Sometimes these defaults can catch new users off guard or make learning/debugging difficult if they are unknown by the user.
```

Particularly about the refetching. While it's great when building no-compromise dynamic UIs, I think it's an overhead in small projects. When consuming (paid) external API, the unnecessary rerenders makes me feel like I'm one bad code away from incurring thousands of dollars of debt.

For small projects, I mainly use `react-query` to hit some API and store the data without the need to set up complicated state management and stuff.

Currently, I disable refetching by default and add a long cacheTime/staleTime.
```
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnMount: false,
      refetchOnWindowFocus: false,
      refetchOnReconnect: false,
    },
  },
})
```
it would be nice if there's a single parameter that would set this up, as I feel like I might be missing some other settings.
Having something like `const queryClient = new QueryClient({passive: true})` would give me peace of mind.

**Comment by TkDodo** [comment, marked as answer]
set `staleTime: Infinity` in the `defaultOptions`.
